test/lib/tasks/bundlers/generateLibraryPreload.js

```diff
@@ -0,0 +1,131 @@
+const {test} = require("ava");
+const path = require("path");
+const chai = require("chai");
+chai.use(require("chai-fs"));
+const assert = chai.assert;
+
+
+const ui5Builder = require("../../../../");
+const builder = ui5Builder.builder;
+const libraryDPath = path.join(__dirname, "..", "..", "..", "fixtures", "library.d");
+const sapUiCorePath = path.join(__dirname, "..", "..", "..", "fixtures", "sap.ui.core");
+
+const recursive = require("recursive-readdir");
+
+const findFiles = (folder) => {
+	return new Promise((resolve, reject) => {
+		recursive(folder, (err, files) => {
+			if (err) {
+				reject(err);
+			} else {
+				resolve(files);
+			}
+		});
+	});
+};
+
+test("Build library.d with library preload", (t) => {
+	const destPath = "./test/tmp/build/library.d/preload";
+	const expectedPath = "./test/expected/build/library.d/preload";
+	const excludedTasks = ["*"];
+	const includedTasks = ["generateLibraryPreload"];
+
+	return builder.build({
+		tree: libraryDTree,
+		destPath,
+		excludedTasks,
+		includedTasks
+	}).then(() => {
+		return findFiles(expectedPath);
+	}).then((expectedFiles) => {
+		// Check for all directories and files
+		assert.directoryDeepEqual(destPath, expectedPath);
+
+		// Check for all file contents
+		expectedFiles.forEach((expectedFile) => {
```

Comment:
This test should use [`t.plan`](https://github.com/avajs/ava/blob/v0.25.0/docs/recipes/when-to-use-plan.md) to define how many files are expected to be compared

Reply:
indeed